File: Linked_List/Singly_Linked_List/ex4_6.java
package Singly_Linked_List;

public class ex4_6 {
    public static void main(String[] args) {
        Node n = new Node(9); // Node n contains item = 9, next = null
        Node second = new Node(8,n);
        Node first = new Node(7,second); // Node first contains item = 8, next = n 

        Node curr = new Node();
        Node prev = new Node();
        curr = first;
        prev = null;
        while (curr != null) {
            if (curr.getItem().equals(7)) {
                System.out.println(curr.getItem());
                System.out.println(prev.getItem());
            }
            else {
                prev = curr;
                curr = curr.getNext();
            }
        }
    }

}

File: Linked_List/Singly_Linked_List/ex4_6.c
#include "C:\\Users\\kritt\\Documents\\Personal_Project\\Data_Structure_and_Algorithm\\Linked_List\\Singly_Linked_List\\Node.h"

int main() {
    struct Node *n = createNode(9, NULL);
    struct Node *second = createNode(8, n);
    struct Node *first = createNode(7, second);

    struct Node *curr;
    struct Node *prev;
    curr = first;
    prev = NULL;

    for(int i=0; i<3; i++) {
        if(getItem(curr) == 8) {
            printf("%d", getItem(curr));
        }
        else {
            prev = curr;
            curr = getNext(curr);
        }
    }
}